### PR TITLE
[DOC] replace dead links with ref or with valid one

### LINKED ISSUE #3526: remove dead links in doc strings
Appears in several other doc strings:

```bash
$  grep -rnw nilearn -e 'nilearn.github.io/manip.*' 

nilearn/reporting/glm_reporter.py:640:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/reporting/glm_reporter.py:645:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/reporting/glm_reporter.py:722:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/reporting/glm_reporter.py:897:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/image/resampling.py:3:See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decoding/decoder.py:85:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decomposition/_base.py:111:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decomposition/_base.py:333:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decomposition/_base.py:386:            See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decomposition/_base.py:479:            See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decomposition/_base.py:554:            See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/decomposition/_multi_pca.py:115:        See http://nilearn.github.io/manipulating_images/input_output.html
nilearn/plotting/img_plotting.py:3:See http://nilearn.github.io/manipulating_images/input_output.html
```

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/3523#discussion_r1112125579_

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3526 

Changes proposed in this pull request:

- replaces links to output/input html page (dead or otherwise) with a `:ref:` where possible
- replaces with just the active link in other places

### CODE REVIEW

**Review by ymzayek — COMMENTED** [review]
LGTM! BTW I remember now why I didn't change some of these. Since these are concerning private modules/functions they will not render in the docs so I thought keeping them is more readable. However, maybe it's not too difficult to find the right document with the sphinx reference heading. And anyways readability of the docstrings is also hindered by standard doc refs in private functions. So maybe it's better to be consistent and cleaner.

**Review by bthirion — COMMENTED** [review]
LGTM otherwise.

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """
 Utilities to resample a Niimg-like object
-See http://nilearn.github.io/manipulating_images/input_output.html
+See http://nilearn.github.io/stable/manipulating_images/input_output.html
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/image/resampling.py:3]: That was the only one I found that was in a module description and not in a parameter description.

That was the only exception I made.

**Comment by Remi-Gau** [comment]
> LGTM! BTW I remember now why I didn't change some of these. Since these are concerning private modules/functions they will not render in the docs so I thought keeping them is more readable. However, maybe it's not too difficult to find the right document with the sphinx reference heading. And anyways readability of the docstrings is also hindered by standard doc refs in private functions. So maybe it's better to be consistent and cleaner.

Yeah I had noticed the pattern but I had found a couple of exceptions, so I kinda opted for a more systematic approach for pretty much the reasons you mentioned.

**Review by ymzayek — APPROVED** [review]
Great then thanks!

**Review by bthirion — APPROVED** [review]
LGTM thx !

**Pull request merged**